src/lightning/data/streaming/combined.py

```diff
@@ -46,7 +46,7 @@ def __init__(
 
     def __len__(self) -> int:
         assert self._weights
-        return int(sum(w * len(d) for w, d in zip(self._weights, self._datasets)))
+        return int(min([1 / w * len(d) for w, d in zip(self._weights, self._datasets) if w > 0]))
```

Comment:
why is this correct? Shouldn't the length correspond to what the iterator is returning?

Reply:
This is an estimation due to the sampling. But the length is the length of the smallest dataset multiplied by the inverse of its weight.

Example. 2 datasets of len 10, 100 and weights of 0.5, 0.5.

At most, we can get 10 * 2 = 20 elements. 10 elements from dataset 1 and 10 from dataset 2.